internal/translation/encryptionatrest/conversion.go

```diff
@@ -0,0 +1,175 @@
+}
+
+func (a *AwsKms) SetSecrets(customerMasterKeyID, roleID string) {
+	a.CustomerMasterKeyID = customerMasterKeyID
+	a.RoleID = roleID
+}
+
+func (a *AwsKms) ToAtlas() *admin.AWSKMSConfiguration {
+	if a == nil {
+		return nil
+	}
+
+	result := &admin.AWSKMSConfiguration{
+		Enabled: a.AwsKms.Enabled,
+		Region:  &a.AwsKms.Region,
+		Valid:   a.AwsKms.Valid,
+
+		RoleId:              &a.RoleID,
+		CustomerMasterKeyID: &a.CustomerMasterKeyID,
+	}
+
+	if result.RoleId == nil && a.CloudProviderIntegrationRole != "" {
+		result.RoleId = &a.CloudProviderIntegrationRole
+	}
+
+	return result
+}
+
+type AzureKeyVault struct {
+	akov2.AzureKeyVault
+	SubscriptionID string
+	KeyVaultName   string
+	KeyIdentifier  string
+	Secret         string
+}
+
+func (az *AzureKeyVault) SetSecrets(subscriptionID, keyVaultName, keyIdentifier, secret string) {
+	az.SubscriptionID = subscriptionID
+	az.KeyVaultName = keyVaultName
+	az.KeyIdentifier = keyIdentifier
+	az.Secret = secret
+}
+
+func (az *AzureKeyVault) ToAtlas() *admin.AzureKeyVault {
+	if az == nil {
+		return nil
+	}
+
+	return &admin.AzureKeyVault{
+		Enabled:           az.AzureKeyVault.Enabled,
+		ClientID:          &az.AzureKeyVault.ClientID,
+		AzureEnvironment:  &az.AzureKeyVault.AzureEnvironment,
+		ResourceGroupName: &az.AzureKeyVault.ResourceGroupName,
+		TenantID:          &az.AzureKeyVault.TenantID,
+
+		SubscriptionID: &az.SubscriptionID,
+		KeyVaultName:   &az.KeyVaultName,
+		KeyIdentifier:  &az.KeyIdentifier,
+		Secret:         &az.Secret,
+	}
+}
+
+type GoogleCloudKms struct {
+	akov2.GoogleCloudKms
+	ServiceAccountKey    string
+	KeyVersionResourceID string
+}
+
+func (g *GoogleCloudKms) SetSecrets(serviceAccountKey, keyVersionResourceID string) {
+	g.ServiceAccountKey = serviceAccountKey
+	g.KeyVersionResourceID = keyVersionResourceID
+}
+
+func (g *GoogleCloudKms) ToAtlas() *admin.GoogleCloudKMS {
+	if g == nil {
+		return nil
+	}
+
+	return &admin.GoogleCloudKMS{
+		Enabled: g.GoogleCloudKms.Enabled,
+
+		ServiceAccountKey:    &g.ServiceAccountKey,
+		KeyVersionResourceID: &g.KeyVersionResourceID,
+	}
+}
+
+func NewEncryptionAtRest(project *akov2.AtlasProject) *EncryptionAtRest {
+	ear := &EncryptionAtRest{}
+	spec := project.Spec.EncryptionAtRest
+	if spec == nil {
+		return nil
+	}
+
+	if *spec.AwsKms.Enabled {
+		ear.AWS.AwsKms = spec.AwsKms
+		for _, role := range project.Status.CloudProviderIntegrations {
+			if role.ProviderName == "AWS" {
+				ear.AWS.CloudProviderIntegrationRole = role.RoleID
+			}
+		}
+	}
+	if *spec.AzureKeyVault.Enabled {
+		ear.Azure.AzureKeyVault = spec.AzureKeyVault
+	}
+	if *spec.GoogleCloudKms.Enabled {
+		ear.GCP.GoogleCloudKms = spec.GoogleCloudKms
+	}
+	return ear
+}
+
+func toAtlas(spec *EncryptionAtRest) *admin.EncryptionAtRest {
+	if spec == nil {
+		return nil
+	}
+
+	return &admin.EncryptionAtRest{
+		AwsKms:         spec.AWS.ToAtlas(),
+		AzureKeyVault:  spec.Azure.ToAtlas(),
+		GoogleCloudKms: spec.GCP.ToAtlas(),
```

Comment:
Why did we chose to go for a conversion method here rather than a function as usual?
I would prefer to stick to toAtlas/fromAtlas functions for consistency.

Reply:
this is fixed